src/main/scala/com/lambtors/poker_api/module/poker/application/create/PokerGameCreator.scala

```diff
@@ -1,18 +1,32 @@
 package com.lambtors.poker_api.module.poker.application.create
 
 import scala.concurrent.{ExecutionContext, Future}
-
-import com.lambtors.poker_api.module.poker.domain.PokerGameRepository
+import com.lambtors.poker_api.module.poker.domain.{PlayerRepository, PokerGameRepository}
 import com.lambtors.poker_api.module.poker.domain.error.PokerGameAlreadyExisting
-import com.lambtors.poker_api.module.poker.domain.model.{AmountOfPlayers, GameId, PokerGame}
+import com.lambtors.poker_api.module.poker.domain.model._
+import com.lambtors.poker_api.module.shared.domain.{DeckProvider, UUIDProvider}
 
-final class PokerGameCreator(pokerGameRepository: PokerGameRepository)(implicit ec: ExecutionContext) {
+final class PokerGameCreator(
+  pokerGameRepository: PokerGameRepository,
+  playerRepository: PlayerRepository,
+  uUIDProvider: UUIDProvider,
+  deckProvider: DeckProvider
+)(implicit ec: ExecutionContext) {
   def create(amountOfPlayers: AmountOfPlayers, gameId: GameId): Future[Unit] =
     pokerGameRepository.search(gameId).flatMap { searchResult =>
       if (searchResult.isDefined) {
         Future.failed[Unit](PokerGameAlreadyExisting(gameId))
       } else {
-        pokerGameRepository.insert(PokerGame(gameId, amountOfPlayers, List.empty))
+        pokerGameRepository.insert(PokerGame(gameId, amountOfPlayers, List.empty)).map(_ => {
+          var cards = deckProvider.provide()
+
+          for (_ <- 1 to amountOfPlayers.amount) {
+            val firstCard :: cardsWithoutFirstCard = cards
```

Comment:
I think we can do directly:

val firstCard :: secondCard :: cardsWithputFirstAndSecondCard = cards

And then we can also try this (but I'm not sure it will work)

val firstCard :: secondCard :: cards = cards